libs/core/src/main/java/org/opensearch/core/transport/TransportMessage.java

```diff
@@ -45,14 +45,24 @@ public abstract class TransportMessage implements Writeable {
 
     private TransportAddress remoteAddress;
 
+    private boolean sampled;
+
     public void remoteAddress(TransportAddress remoteAddress) {
         this.remoteAddress = remoteAddress;
     }
 
+    public void markResponseAsSampled() {
```

Comment:
@nishchay21 we discussed that on the call and in the issue itself, the tracing details should entangle with OpenSearch core, this is no go from the start. If that could not be done with OTEL, we need to find the right framework and not make OpenSearch core to be one.

[1] https://github.com/opensearch-project/OpenSearch/issues/12315#issuecomment-2015037013

Reply:
@reta 

We do not plan to store this information. However I would like to add that a basic/minimal stitch will be required to read these header information within the core. For now we plan to just store the headers of the response only within the transport message which will not affect the basic working of OpenSearch core and OTel plugin will sample outside the scope of core itself.

Reply:
@nishchay21 headers are fine, check also https://opentelemetry.io/docs/concepts/signals/baggage/ for the way OTEL uses propagation between spans